### PR TITLE
Cluster Profiles SPA - basic skeleton, models and endpoints to perform CRUD operations

### PR DESCRIPTION
**Pull request by viraj2712**
#6045

### CODE REVIEW

File: server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts [hunk 1]
```diff
@@ -241,4 +241,11 @@ export default class {
   static apiAdminAccessTokenRevokePath(id: number) {
     return `${this.apiAdminAccessTokensBasePath()}/${id}/revoke`;
   }
+
+  static apiAdminAccessClusterProfilesPath(id?: string) {
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts]: any reason for a single method for `GET` and `INDEX`?

File: server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  dummy?: ClusterProfiles;
+}
+
+export class ClusterProfilesPage extends Page<null, State> {
+  componentToDisplay(vnode: m.Vnode<null, State>): m.Children {
+    return <ClusterProfilesWidget/>;
+  }
+
+  pageName(): string {
+    return "SPA Name goes here!";
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx]: fix this?

File: server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* limitations under the License.
+*/
+
+import * as m from "mithril";
+import {ClusterProfiles} from "models/cluster_profiles/cluster_profiles";
+import {ClusterProfilesWidget} from "views/pages/cluster_profiles/cluster_profiles_widget";
+import {Page} from "views/pages/page";
+
+interface State {
+  dummy?: ClusterProfiles;
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx]: fix this?

File: server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    return <ClusterProfilesWidget/>;
+  }
+
+  pageName(): string {
+    return "SPA Name goes here!";
+  }
+
+  fetchData(vnode: m.Vnode<null, State>): Promise<any> {
+    // to be implemented
+    return Promise.resolve();
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles.tsx]: fix this?

File: server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles/cluster_profiles_widget.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import {MithrilViewComponent} from "jsx/mithril-component";
+import * as m from "mithril";
+import {ClusterProfiles} from "models/cluster_profiles/cluster_profiles";
+
+interface Attrs {
+  dummy?: ClusterProfiles;
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/rails/webpack/views/pages/cluster_profiles/cluster_profiles_widget.tsx]: fix this?

File: spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/ClusterProfilesController.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  @Override
+  public void setupRoutes() {
+     path(controllerBasePath(), () -> {
+        before("", authenticationHelper::checkAdminUserAnd403);
+        get("", this::index, engine);
+    });
+  }
+  public ModelAndView index(Request request, Response response) {
+      Map<Object, Object> object = new HashMap<Object, Object>() {{
+          put("viewTitle", "ClusterProfiles");
```
Comment (GaneshSPatil) [inline spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/ClusterProfilesController.java]: how about `Cluster Profiles`?

File: spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/ClusterProfilesController.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import com.thoughtworks.go.spark.spring.SPAAuthenticationHelper;
+import spark.ModelAndView;
+import spark.Request;
+import spark.Response;
+import spark.TemplateEngine;
+import java.util.HashMap;
+import java.util.Map;
+import static spark.Spark.*;
+
+public class ClusterProfilesController implements SparkController {
```
Comment (GaneshSPatil) [inline spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/ClusterProfilesController.java]: how about adding security tests for this controller?

File: server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts (hunk 1 above)
Comment (viraj2712) [reply server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts]: No specific reason. Few have separate methods for `GET` and `INDEX` few have a single method. Which approach should we follow?

**Comment by viraj2712** [comment]
@GaneshSPatil We have removed all the files which are not implemented yet.

**Pull request merged**